### PR TITLE
Basic integration tests added.

### PR DESCRIPTION
**Pull request by panasevychol**
Basic integration tests added.

### CODE REVIEW

File: tests/integration/generic/fetchd_client.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            self._make_entrypoint(tmpdir)
+            volumes = {tmpdir: {"bind": self.MOUNT_PATH, "mode": "rw"}}
+            entrypoint = os.path.join(self.MOUNT_PATH, self.ENTRYPOINT_FILENAME)
+            self.container = self.client.containers.run(
+                self.IMG_TAG,
+                detach=True,
+                # network="host",
+                volumes=volumes,
+                entrypoint=str(entrypoint),
+                name=self.CONTAINER_NAME,
```
[the remark this answers is not in the document]
Comment (MissingNO57) [reply tests/integration/generic/fetchd_client.py]: Let's give it a try

File: tests/integration/generic/fetchd_client.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    MINIMUM_DOCKER_VERSION = (19, 0, 0)
+    ENDPOINT = "http://127.0.0.1:1317"
+
+    IMG_TAG = "fetchai/fetchd:0.8.4"
+    ENTRYPOINT_FILENAME = "entry.sh"
+    MOUNT_PATH = "/mnt"
+    PORTS = {9090: 9090, 1317: 1317, 26657: 26657}
+
+    DENOM = "stake"
+    VALIDATOR_ADDRESS = Address(
```
Comment (5A11) [inline tests/integration/generic/fetchd_client.py:50]: Retrieve from `generic/config.py`

File: tests/integration/generic/fetchd_client.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+    MINIMUM_DOCKER_VERSION = (19, 0, 0)
+    ENDPOINT = "http://127.0.0.1:1317"
+
+    IMG_TAG = "fetchai/fetchd:0.8.4"
+    ENTRYPOINT_FILENAME = "entry.sh"
+    MOUNT_PATH = "/mnt"
+    PORTS = {9090: 9090, 1317: 1317, 26657: 26657}
+
+    DENOM = "stake"
```
Comment (5A11) [inline tests/integration/generic/fetchd_client.py:49]: Retrieve from `generic/config.py` above

File: tests/integration/generic/config.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+# Node config
+GRPC_ENDPOINT_ADDRESS = "localhost:9090"
+REST_ENDPOINT_ADDRESS = "http://localhost:1317"
+CHAIN_ID = "testing"
+
+# Private key of sender account
+VALIDATOR_PK = PrivateKey(
+    bytes.fromhex("0ba1db680226f19d4a2ea64a1c0ea40d1ffa3cb98532a9fa366994bb689a34ae")
+)
+VALIDATOR_ADDRESS = str(Address(VALIDATOR_PK))
```
Comment (5A11) [inline tests/integration/generic/config.py:44]: I think:

```suggestion
VALIDATOR_ADDRESS = Address(VALIDATOR_PK)
```

File: tests/integration/generic/fetchd_client.py [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+from cosmpy.common.rest_client import RestClient
+from cosmpy.crypto.address import Address
+from cosmpy.crypto.keypairs import PrivateKey
+
+
+class FetchdDockerImage:
+    """Class to operate Fetchd node with Docker engine."""
+
+    MINIMUM_DOCKER_VERSION = (19, 0, 0)
+    ENDPOINT = "http://127.0.0.1:1317"
```
Comment (5A11) [inline tests/integration/generic/fetchd_client.py:42]: Retrieve `REST_ENDPOINT` from generic/config.py above

File: tests/integration/generic/config.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+# Private key of recipient account
+BOB_PK = PrivateKey(
+    bytes.fromhex("439861b21d146e83fe99496f4998a305c83cfbc24717c77e32b06d224bf1e636")
+)
+BOB_ADDRESS = str(Address(BOB_PK))
```
Comment (5A11) [inline tests/integration/generic/config.py:50]: Same as above (I think). (see the diffs in its usage).

File: tests/integration/generic/test_cases.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    """Base test case for Fetchd node."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Set up Fetchd node for testing."""
+        cls.client = FetchdDockerImage()
+        cls._try_launch_image()
+
+    @classmethod
+    def _try_launch_image(cls):
```
Comment (5A11) [inline tests/integration/generic/test_cases.py:39]: Add comment why we are doing this (say first run fails in CI)

**Pull request merged**